src/Compilers/VisualBasic/Portable/Symbols/Metadata/PE/TupleTypeDecoder.vb

```diff
@@ -110,14 +110,20 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
                 ' the type. Bad metadata.
             End Try
 
+            If metadataType.GetUseSiteErrorInfo() IsNot Nothing Then
+                Return metadataType
+            End If
+
             ' Bad metadata
             Return New UnsupportedMetadataTypeSymbol()
         End Function
 
         Private Function DecodeType(type As TypeSymbol) As TypeSymbol
             Select Case type.Kind
-                Case SymbolKind.ErrorType,
-                    SymbolKind.DynamicType,
+                Case SymbolKind.ErrorType
+                    Return DecodeNamedType(DirectCast(type, NamedTypeSymbol))
```

Comment:
Is the same change needed in C#?

Reply:
The PR is marked for personal review. Still working on it.

Reply:
I'll be removing this change.